lua/apisix/plugins/limit-req.lua

```diff
@@ -22,28 +22,34 @@ end
 
 
 function _M.access(conf, ctx)
-    local limit_ins = core.lrucache.plugin_ctx(plugin_name, ctx,
+    local lim, err = core.lrucache.plugin_ctx(plugin_name, ctx,
                                                create_limit_obj, conf)
 
-    local key = ctx.var[conf.key]
-    if not key or key == "" then
-        key = ""
-        core.log.warn("fetched empty string value as key to limit the request ",
-                      "maybe wrong, please pay attention to this.")
+    if not lim then
+        core.log.error("failed to instantiate a resty.limit.req object: ", err)
+        return ngx.HTTP_INTERNAL_SERVER_ERROR
+    end
+
+    if conf.key ~= 'remote_addr' then
```

Comment:
need to delete this logic, we can check the argument at https://github.com/iresty/apisix/pull/52/files#diff-bf9754cd6cd61a072ce919164c35ad16R13, that is a todo.